ipa-core/src/protocol/ipa_prf/shuffle/sharded.rs

```diff
@@ -288,11 +265,11 @@ where
     // shared with the left helper.
     let x2: Vec<S::Share> = ctx
         .narrow(&ShuffleStep::Permute31)
-        .mask_and_shuffle(Direction::Left, &x1)
+        .mask_and_shuffle(Direction::Left, x1.iter().copied())
```

Comment:
With `ShuffleShare` restricted to `BooleanArray`, it acquires `Copy`, which makes the `Borrow` generic on `mask_and_shuffle` that I recently added unnecessary.